src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -0,0 +1,488 @@
+	PcapAPIURL         string   `short:"u" long:"pcap-api-url" description:"The URL of the PCAP API, e.g. pcap.cf.$LANDSCAPE_DOMAIN" env:"PCAP_API" required:"true"`
+	Filter             string   `short:"f" long:"filter" description:"Allows to provide a filter expression in pcap filter format." required:"false"`
+	Interface          string   `short:"i" long:"interface" description:"Specifies the network interface to listen on." default:"eth0" required:"false"`
+	Type               string   `short:"t" long:"type" description:"Specifies the type of process to capture for the app." default:"web" required:"false"`
+	BoshConfigFilename string   `short:"c" long:"bosh-config" description:"Path to the BOSH config file, used for the UAA Token" default:"${HOME}/.bosh/config" required:"false"`
+	BoshEnvironment    string   `short:"e" long:"bosh-environment" description:"The BOSH environment to use for retrieving the BOSH UAA token from the BOSH config file" default:"bosh" required:"false"`
+	Deployment         string   `short:"d" long:"deployment" description:"The name of the deployment in which you would like to capture." required:"true"`
+	InstanceGroups     []string `short:"g" long:"instance-group" description:"The name of an instance group in the deployment in which you would like to capture. Can be defined multiple times." required:"true"`
+	InstanceIds        []string `positional-arg-name:"ids" description:"The instance IDs of the deployment to capture." required:"false"`
+	SnapLength         uint16   `short:"l" long:"snaplen" description:"Snap Length, defining the captured length of the packet, with the remainder truncated. The real packet length is recorded." default:"65535"`
+	Verbose            bool     `short:"v" long:"verbose" description:"Show verbose debug information"`
+	Quiet              bool     `short:"q" long:"quiet" description:"Show only warnings and errors"`
+}
+
+// init sets up the zap.Logger. Currently outputs to stderr in Console format.
+func init() {
+	atomicLogLevel = zap.NewAtomicLevelAt(zap.InfoLevel)
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	outputPaths := []string{"stderr"}
+	zapConfig := zap.Config{
+		Level:             atomicLogLevel,
+		DisableCaller:     true,
+		DisableStacktrace: true,
+		Encoding:          "console",
+		EncoderConfig:     encoderConfig,
+		OutputPaths:       outputPaths,
+		ErrorOutputPaths:  outputPaths,
+	}
+	logger = zap.Must(zapConfig.Build())
+	zap.ReplaceGlobals(logger)
+	logger.Debug("successfully set up logger", zap.Strings("output-paths", outputPaths))
+}
+
+func main() {
+	var (
+		err         error
+		opts        options
+		apiURL      *url.URL
+		client      *pcap.Client
+		environment *Environment
+	)
+
+	// Parse command-line arguments
+	_, err = flags.Parse(&opts)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	defer func() {
```

Comment:
This was on purpose. I didn't want to have double outputs of the flags.Parse error, because flags.Parse already prints the error message itself.

I couldn't figure out in reasonable time how to distinguish the flags error cleanly enough to not print it conditionally.

I would add a comment on the first exit to explain why it's there.

Reply:
Nevermind, figured it out. Will be in the next commit.